02_knight_and_princess/setyaev_alexander/graph.cpp

```diff
@@ -0,0 +1,196 @@
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+class Vertex {
+ public:
+  Vertex(int id);
+  int get_id() const;
+  std::vector<int> connected_edges;
+
+ private:
+  int _id = 0;
+};
+
+struct Edge {
+ public:
+  Edge(int id, int first_vertex_id, int second_vertex_id);
+  int get_id() const;
+  int get_first_id() const;
+  int get_second_id() const;
+
+ private:
+  int _id = 0;
+  int _first_vertex_id = 0;
+  int _second_vertex_id = 0;
+};
+
+class Graph {
+ public:
+  void add_vertex();
+  void add_edge(int first_vertex_id, int second_vertex_id);
+  std::string to_string() const;
+  bool is_vertex_exists(int id) const;
+  bool is_edge_exists(int id) const;
+  int get_new_vertex_id() const;
+  int get_new_edge_id() const;
+
+ private:
+  std::vector<Vertex> _vertices;
+  std::vector<Edge> _edges;
+};
+
+int Vertex::get_id() const {
+  return _id;
+}
+int Edge::get_id() const {
+  return _id;
+}
+
+int Edge::get_first_id() const {
+  return _first_vertex_id;
+}
+
+int Edge::get_second_id() const {
+  return _second_vertex_id;
+}
+
+Edge::Edge(int id, int first_vertex_id, int second_vertex_id) {
+  _id = id;
+  _first_vertex_id = first_vertex_id;
+  _second_vertex_id = second_vertex_id;
+}
+
+Vertex::Vertex(int id) {
+  _id = id;
+}
+
+void Graph::add_vertex() {
+  _vertices.emplace_back(get_new_vertex_id());
+}
+
+int Graph::get_new_vertex_id() const {
+  return _vertices.size();
```

Comment:
`id` не имеет прямой связи с индексом. Нужно завести специальный счетчик
```cpp
VertexId vertex_id_counter_ = 0;
```
```suggestion
  return vertex_id_counter_++;
```